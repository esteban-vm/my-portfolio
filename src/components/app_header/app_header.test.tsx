import { render, screen } from '@/tests'
import AppHeader from './app_header'

describe(`${AppHeader.name}:`, () => {
  it('should render correctly', () => {
    const { asFragment } = render(<AppHeader />)
    expect(asFragment()).toMatchSnapshot()
    expect(screen.getByRole('banner')).toBeInTheDocument()
  })
})
