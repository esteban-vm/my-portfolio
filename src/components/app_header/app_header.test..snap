// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppHeader: > should render correctly 1`] = `
<DocumentFragment>
  <header
    aria-labelledby=":r0:"
    class="container pointer-events-none absolute left-1/2 top-3 z-30 flex h-[11vh] -translate-x-1/2 md:h-[12vh] lg:h-[13vh]"
  >
    <div
      class="relative flex size-full items-center justify-around"
    >
      <h1
        id=":r0:"
      >
        <a
          class="pointer-events-auto animate-glitch text-center font-saiba45 text-neon-yellow ~text-4xl/7xl logo-shadow hover:animate-none"
          href="/"
        >
          Esteban
          <span
            class="hidden font-saiba45 md:inline"
          >
            Â V.M.
          </span>
        </a>
      </h1>
      <button
        class="pointer-events-auto flex items-center justify-center text-neon-yellow lg:hidden [&>svg]:~size-8/10 active:[&>svg]:scale-90"
        type="button"
      >
        <svg
          aria-label="Open Menu"
          fill="none"
          height="1em"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <line
            x1="4"
            x2="20"
            y1="12"
            y2="12"
          />
          <line
            x1="4"
            x2="20"
            y1="6"
            y2="6"
          />
          <line
            x1="4"
            x2="20"
            y1="18"
            y2="18"
          />
        </svg>
      </button>
      <nav
        class="top-[-500%] animate-out slide-out-to-top absolute flex w-[95%] flex-col items-center justify-around text-nowrap bg-black/70 text-center duration-700 fill-mode-forwards nav-height-sm md:nav-height-md lg:static lg:size-auto lg:animate-none lg:flex-row lg:gap-8 lg:bg-transparent"
      >
        <a
          class="pointer-events-auto relative inline-block font-bold uppercase text-neon-yellow ~text-sm/base after:bottom-0 after:left-0 after:h-1 after:w-full after:origin-bottom-right after:scale-x-0 after:bg-neon-yellow after:transition-transform after:duration-200 after:ease-out after:content-[''] hover:opacity-90 hover:after:origin-bottom-left hover:after:scale-x-100 active:scale-95 lg:after:absolute"
          href="/about"
          id="9d1222e7-ad5f-4506-b583-2ea9cfe989d8"
        >
          About Me
        </a>
        <a
          class="pointer-events-auto relative inline-block font-bold uppercase text-neon-yellow ~text-sm/base after:bottom-0 after:left-0 after:h-1 after:w-full after:origin-bottom-right after:scale-x-0 after:bg-neon-yellow after:transition-transform after:duration-200 after:ease-out after:content-[''] hover:opacity-90 hover:after:origin-bottom-left hover:after:scale-x-100 active:scale-95 lg:after:absolute"
          href="/projects"
          id="6b3e6de5-f39b-4b15-b744-9c12d7710dd8"
        >
          My Projects
        </a>
        <a
          class="pointer-events-auto relative inline-block font-bold uppercase text-neon-yellow ~text-sm/base after:bottom-0 after:left-0 after:h-1 after:w-full after:origin-bottom-right after:scale-x-0 after:bg-neon-yellow after:transition-transform after:duration-200 after:ease-out after:content-[''] hover:opacity-90 hover:after:origin-bottom-left hover:after:scale-x-100 active:scale-95 lg:after:absolute"
          href="/contact"
          id="c2a12051-a97a-4e0b-ae14-4e63098a3bb2"
        >
          Contact Me
        </a>
      </nav>
    </div>
  </header>
</DocumentFragment>
`;
